R/surv-brier.R

```diff
@@ -0,0 +1,166 @@
+#' Brier score for right censored data
+#'
+#' Compute the time-dependent Brier score for right censored data. Which is the
+#' mean squared error at time point `eval_time`.
+#'
+#' @family dynamic survival metrics
+#' @templateVar fn brier_survival
+#' @template return-dynamic-survival
+#' @details
+#'
+#' This method will automatically group by the `eval_time` argument.
+#'
+#' Smaller values of the score are associated with better model performance.
+#'
+#' @inheritParams pr_auc
+#'
+#' @param data A `data.frame` containing the columns specified by `truth` and
+#' `estimate`.
+#'
+#' @param truth The column identifier for the true class survival result (that
+#' is created using [survival::Surv()].). This should be an unquoted column name
+#' although this argument is passed by expression and supports
+#' [quasiquotation][rlang::quasiquotation] (you can unquote column names). For
+#' `_vec()` functions, an [survival::Surv()] object.
+#'
+#' @param estimate The column identifier for the survival probabilities. This
+#' should be a numeric vector. This should be an unquoted column name although
+#' this argument is passed by expression and supports
+#'  [quasiquotation][rlang::quasiquotation] (you can unquote column names). For
+#' `_vec()` functions, a numeric vector.
+#'
+#' @param censoring_weights The column identifier for censoring weights. This is
+#' expected to a numeric vector. This should be an unquoted column name although
+#' this argument is passed by expression and supports
+#' [quasiquotation][rlang::quasiquotation] (you can unquote column names). For
+#' `_vec()` functions, a numeric vector.
+#'
+#' @param eval_time The column identifier for the time point. This
+#' should be a numeric vector, with 1 unique value for each group. This should
+#' be an unquoted column name although this argument is passed by expression and
+#' supports [quasiquotation][rlang::quasiquotation] (you can unquote column
+#' names). For `_vec()` functions, a numeric vector.
+#'
+#' @param ... Not currently used.
```

Comment:
we could back that up with a `rlang::check_dots_empty()`

Reply:
We are doing that here https://github.com/tidymodels/yardstick/blob/main/R/template.R#L329

Reply:
ah, nice. then we should pass the dots from `brier_survival.data.frame()` to `dynamic_survival_metric_summarizer()` though